libsolidity/parsing/Token.cpp

```diff
@@ -80,8 +112,35 @@ char const Token::m_tokenType[] =
 {
 	TOKEN_LIST(KT, KK)
 };
-Token::Value Token::fromIdentifierOrKeyword(const std::string& _name)
+tuple<Token::Value, string> Token::fromIdentifierOrKeyword(const string& _literal)
 {
+	string token = _literal;
+	string details;
+	if (_literal == "uintM" || _literal == "intM" || _literal == "fixedMxN" || _literal == "ufixedMxN" || _literal == "bytesM")
+		return make_pair(Token::Identifier, details);
+	if (_literal.find_first_of("0123456789") != string::npos)
+	{
+		string baseType = _literal.substr(0, _literal.find_first_of("0123456789"));
```

Comment:
Please store the position and check that it is not too large. `stoi` below should not throw.